File: C/DAY2/factorial.c
#include<stdio.h>
int main(){

    //factorial of number given by user
    int n;
    int fact=1;
    printf("enter a natural number\n");
    scanf("%d",&n);
    for(int i=1;i<=n;i++){
        fact =fact * i; //or fact *=i;
    }
    printf("factorial of %d is %d\n",n,fact);
   

}

File: EclipseWorkspace.java/Code_Interview/JavaInterview_2_8_25/src/javaInterview/Factorial.java
package javaInterview;

import java.util.Scanner;

// How to calculate factorial using recursion ?
public class Factorial {

	public static void main(String[] args) {
		
		int fact=1,n;
		
		System.out.println("enter a number");
		Scanner sc = new Scanner(System.in);
		n = sc.nextInt();
		
		for(int i=1;i<=n;i++) {
			fact *= i;
		}
		System.out.println("factorial of " +n+ " is = "+fact);
		
	}
}
